shared/src/hooks/useFetchEarnVaultData.ts

```diff
@@ -131,6 +136,22 @@ const useFetchEarnVaultData = (): V2VaultData => {
           (_vaultState as { round?: number }).round ? _vaultState : { round: 1 }
         ) as { round: number };
 
+        // we use roundPricePerShare for earn treasury because pricePerShare goes down
+        // during the round itself
+        const roundPricePerSharePromise = vaultState.round
+          ? contract.roundPricePerShare(vaultState.round - 1)
+          : Promise.resolve(BigNumber.from(0));
+
+        let roundPricePerShare: BigNumber = BigNumber.from(0);
+        await roundPricePerSharePromise.then((v) => (roundPricePerShare = v));
```

Comment:
Could we write this as something like `const roundPricePerShare = vaultState.round ? await contract.roundPricePerShare(vaultState.round - 1) : BigNumber.from(0)`?